File: Lemon Game Store/Game.cs
using System;
using System.Collections.Generic; //for the list
using System.Linq; //for the sum

namespace Lemon_Game_Store
{
    class Game : BaseDetails
    {
        public static List<Game> gameList = new List<Game>(); // public static in order to become available globally
        public static List<double> gameCart = new List<double>(); //double is used because thats the type of Price
        private string _platform { get; set; } = "PC";  // removed from Basedetail.cs cheanged it to private
        public override void CreateList()
        {
            gameList.Add(new Game { Title = "PERSONA 5", Condition = "Brandnew", Price = 1499, _platform = "Console" });

            gameList.Add(new Game { Title = "UNCHARTED", Condition = "2nd Hand", Price = 500, _platform = "Console" });

            gameList.Add(new Game { Title = "WITCHER 3", Condition = "Brandnew", Price = 1999, _platform = "PC" });
        }
        public static void SellGame()
        {
            foreach (var games in gameList)
            {
                Console.WriteLine($"Title: {games.Title}");
                Console.WriteLine($"Platform: {games._platform}");
                Console.WriteLine($"Item Condition: {games.Condition}");
                Console.WriteLine($"Price: {games.Price}");
                Console.WriteLine("-----------------------------");
            }
            Console.WriteLine("What game to buy?");
            string upperAnswer = Console.ReadLine();
            string userGameChoice = upperAnswer.ToUpper();
            bool searchFlag = false;

            while (userGameChoice != "0")
            {
                foreach (var games_ in gameList)
                {
                    if (games_.Title == userGameChoice)
                    {
                        Console.WriteLine("Search is successful");
                        Console.WriteLine($"The price is {games_.Price}");
                        gameCart.Add(games_.Price);
                        Console.WriteLine("The game was added to cart, anything else? (Enter 0 to proceed to payment)");
                        string upper = Console.ReadLine();
                        userGameChoice = upper.ToUpper();
                        searchFlag = false;
                        break;
                    }
                    else
                    {
                        searchFlag = true;
                    }
                }
                if (searchFlag == true)
                {
                    Console.WriteLine("Search is unsuccessful, it seems like we don't have that");
                    Console.WriteLine("Search another game (Enter 0 to exit)");
                    string upper = Console.ReadLine();
                    userGameChoice = upper.ToUpper();
                }
            }
            var totalAmount = gameCart.Sum();
            Console.WriteLine($"The total amount to pay is {totalAmount}");
            Console.WriteLine("Input the amount you have: ");
            int userMoney = Convert.ToInt32(Console.ReadLine());
            var exchange = userMoney - totalAmount;
            if (totalAmount > userMoney)
            {
                Console.WriteLine("Not enough funds!");
                Console.WriteLine("Try again?(Y/N)");
                string upper = Console.ReadLine();
                string userAns = upper.ToUpper();
                if (userAns == "Y")
                {
                    Console.WriteLine("Input the amount you have: ");
                    int userMoney_ = Convert.ToInt32(Console.ReadLine());
                    while (userMoney_ < totalAmount)
                    {
                        Console.WriteLine("Not enough funds!");
                        Console.WriteLine("Input the amount you have: ");
                        userMoney_ = Convert.ToInt32(Console.ReadLine());
                    }
                    var exchange_ = userMoney_ - totalAmount;
                    Console.WriteLine($"Thank you! Your exchange is {exchange_}");
                }
            }
            else
            {
                Console.WriteLine($"Thank you! Your exchange is {exchange}");
            }
        }
 
    }
}

File: Python Lemon Game Store/Game.py
from OOP_FINALS import Item as I
class Game:
    def __init__(self):
        self.sell_product = []
        self.buy_product = []
        self.sell_cart = []
        self.buy_cart = []
    def createlist_sellproduct(self):
        self.sell_product.append(I.Item(Title='Persona 5', Platform='Console', Condition='Brandnew', Price=1499))
        self.sell_product.append(I.Item(Title='Uncharted', Platform='Console', Condition='Brandnew', Price=500))
        self.sell_product.append(I.Item(Title='Witcher 3', Platform='Console', Condition='Brandnew', Price=1999))
    def sellproduct(self):
        for product in self.sell_product:
            print("Title: {}\nPlatform: {}\nCondition: {}\nPrice: {}".format(product.Title,
                                                                             product.Platform,
                                                                             product.Condition,
                                                                             product.Price))
            print("-----------------------------")
        user_input = input("Choose what to sell:")
        searchFlag = False
        while user_input != "0":
            for products in self.sell_product:
                if user_input.lower() == products.Title.lower():
                    print(products.Price)
                    self.sell_cart.append(products.Price)
                    user_input = input("The game was added to cart, anything else? (Enter 0 to proceed to payment)")
                    searchFlag = False
                    break
                else:
                    searchFlag = True
            if searchFlag == True:
                print("Search is unsuccessful, It seems like we don't have that");
                user_input = input("Search another game (Enter 0 to proceed to payment): ")
        total_amount_to_pay = sum(self.sell_cart)
        customer_pay = int(input("Input the customer's money: "))
        if customer_pay >= total_amount_to_pay:
            change = customer_pay - total_amount_to_pay
            print("Transaction Successful")
            print("Change: ", change)
        elif customer_pay < total_amount_to_pay:
            print("Insufficient Money")


    def createlist_buyproduct(self):
        self.buy_product.append(I.Item(Title='Sekiro', Platform='PC', Condition='2nd Hand', Price=1899))
        self.buy_product.append(I.Item(Title='Doom Eternal', Platform='PC', Condition='2nd Hand', Price=1380))
        self.buy_product.append(I.Item(Title='Bayonetta', Platform='Console', Condition='2nd Hand', Price=250))

    def buyproduct(self):
        for product in self.buy_product:
            print("Title: {}\nPlatform: {}\nCondition: {}\nPrice: {}".format(product.Title,
                                                                             product.Platform,
                                                                             product.Condition,
                                                                             product.Price))
            print("-----------------------------")
        user_input = input("Choose what to buy:")
        searchFlag = False
        while user_input != "0":
            for products in self.buy_product:
                if user_input.lower() == products.Title.lower():
                    print(products.Price)
                    self.buy_cart.append(products.Price)
                    self.sell_product.append(products)
                    user_input = input("The game was added to cart, anything else? (Enter 0 to proceed to payment)")
                    searchFlag = False
                    break
                else:
                    searchFlag = True
            if searchFlag == True:
                print("Search is unsuccessful, It seems like we don't want to buy that");
                user_input = input("Search another game (Enter 0 to proceed to payment): ")

        total_amount_to_pay = sum(self.buy_cart)
        cashier_pay = int(input("Input the cashier's money: "))
        if cashier_pay >= total_amount_to_pay:
            change = cashier_pay - total_amount_to_pay
            print("Transaction Successful")
            print("Change: ", change)
        elif cashier_pay < total_amount_to_pay:
            print("Insufficient Money")


class Gameconsole(Game): # inherited the Game class

    def createlist_sellproduct(self):
        self.sell_product.append(I.Item(Title="PS5", Condition="Brandnew", Price=27650, Platform=""))
        self.sell_product.append(I.Item(Title="PS4", Condition="Brandnew", Price=8000, Platform=""))
        self.sell_product.append(I.Item(Title="Nintendo Switch", Condition="Brandnew", Price=14999, Platform=""))

    def createlist_buyproduct(self):
        self.buy_product.append(I.Item(Title="PS3", Condition="2nd Hand", Price=8500, Platform=""))
        self.buy_product.append(I.Item(Title="PS2", Condition="2nd Hand", Price=3000, Platform=""))
        self.buy_product.append(I.Item(Title="Xbox 360", Condition="2nd Hand", Price=5000, Platform=""))

